core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunction.java

```diff
@@ -49,6 +49,16 @@
      *            the value which decay function should take once the distance
      *            reaches this scale
      * */
-    public double processScale(double scale, double decay);
+    public abstract double processScale(double scale, double decay);
 
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
this method impl doesn't make sense to me and it doesn't have a hashCode either why don't you just leave it as an interface?